File: backend/models/floor_plan.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class FloorPlanRecord:
    floor_plan_id: str               # Firestore document ID
    user_id: str                     # owning Security Manager UID
    name: str                        # e.g. "Level 1 — Warehouse A"
    url: str                         # Firebase Storage download URL
    uploaded_at: str                 # ISO 8601
    is_active: bool                  # only one plan active per user at a time
    scale_pixels_per_meter: Optional[float] = None  # set after calibration, None until calibrated

File: frontend/types/floorPlan.ts
export interface FloorPlanRecord {
  floor_plan_id: string;
  user_id: string;
  name: string;
  url: string;                           // Firebase Storage download URL
  storage_path: string;                  // e.g. floor_plans/uid/timestamp_file.png
  uploaded_at: string;                   // ISO 8601
  is_active: boolean;
  scale_pixels_per_meter: number | null; // null until calibrated
}
